src/codegen/ast_interpreter.cpp

```diff
@@ -1013,38 +1067,26 @@ Value ASTInterpreter::visit_str(AST_Str* node) {
 }
 
 Value ASTInterpreter::visit_name(AST_Name* node) {
-    switch (node->lookup_type) {
-        case AST_Name::UNKNOWN: {
-            ScopeInfo::VarScopeType vst = scope_info->getScopeTypeOfName(node->id);
-            if (vst == ScopeInfo::VarScopeType::GLOBAL) {
-                node->lookup_type = AST_Name::GLOBAL;
-                return getGlobal(source_info->parent_module, &node->id.str());
-            } else if (vst == ScopeInfo::VarScopeType::DEREF) {
-                node->lookup_type = AST_Name::CLOSURE;
-                return getattr(passed_closure, node->id.c_str());
-            } else {
-                bool is_old_local = (vst == ScopeInfo::VarScopeType::NAME);
-                node->lookup_type = is_old_local ? AST_Name::LOCAL : AST_Name::FAST_LOCAL;
-
-                SymMap::iterator it = sym_table.find(node->id);
-                if (it != sym_table.end()) {
-                    Box* value = it->second;
-                    return value;
-                }
-
-                // classdefs (and some other cases like eval) have different scoping rules than functions:
-                if (is_old_local)
-                    return getGlobal(source_info->parent_module, &node->id.str());
-
-                assertNameDefined(0, node->id.c_str(), UnboundLocalError, true);
-                return Value();
-            }
+    if (node->lookup_type == AST_Name::UNKNOWN) {
+        ScopeInfo::VarScopeType vst = scope_info->getScopeTypeOfName(node->id);
+        if (vst == ScopeInfo::VarScopeType::GLOBAL) {
+            node->lookup_type = AST_Name::GLOBAL;
+        } else if (vst == ScopeInfo::VarScopeType::DEREF) {
+            node->lookup_type = AST_Name::DEREF;
+            return getattr(passed_closure, node->id.c_str());
```

Comment:
ok this is really tiny, but this inconsistency (having only one branch immediately fetch the name) makes me think that something was different about this path, but then I looked further down and it looks like they're the same.